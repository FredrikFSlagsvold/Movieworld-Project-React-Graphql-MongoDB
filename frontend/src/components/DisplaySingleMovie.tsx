import { gql, useQuery } from '@apollo/client';
import Box from '@mui/material/Box';

type DisplaySingleMovieProps ={
    poster_path: String;
    original_language: String;
    title: String;
    runtime: number;
    genres: [String]; 
}


export default function DisplaySingleMovie({poster_path, original_language, title, runtime, genres}: DisplaySingleMovieProps){
    let genresString = genres.join(', ');
    // while (genresString.length<28)

    /*
    <div style={{ display: "flex", justifyContent: "space-between"}}>
                <p style={{padding:"0", marginBottom:"0", marginTop:"0"}}>{runtime} min</p>
                <p style={{padding:"0", marginBottom:"0", marginTop:"0"}}>{original_language} </p>
            </div>
    */
    



    return <>
    <div style={{
        width: 190,
        height: 370,
        padding: "48px",
        backgroundColor: "white",
        fontFamily: "Verdana, sans-serif, Areal",
        fontSize:"12px"
        
    }} >
        <img src={"https://image.tmdb.org/t/p/original/"+ poster_path} width="190px" height="284.8px"/>
            <div style={{textAlign:"center"}}>
            {(() => {
        if (title.length>24) {
          return (
            <div><strong>{title.substring(0,21)+ "..."}</strong></div>
          )
        } else {
          return (
            <div><strong>{title}</strong></div>
          )
        }
      })()}
            </div>
            
            <div style={{textAlign:"center"}}>
            {(() => {
                if (genresString.length>25) {
                    return (
                        <div>{genresString.substring(0,23)+ "..."}</div>
                    )
                } else {
                    return (
                        <div>{genresString}</div>
                    )
                }
            })()}
            </div>
    </div>
    </>
}