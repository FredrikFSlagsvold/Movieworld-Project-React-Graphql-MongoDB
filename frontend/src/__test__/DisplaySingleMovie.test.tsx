import renderer from 'react-test-renderer';
import DisplaySingleMovie from '../components/DisplaySingleMovie';



it('renders when ', () => {
  const tree = renderer.create(<DisplaySingleMovie poster_path="/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg" original_language="English" title="Godzilla vs. Kong" runtime={113} genres={["Action"]}/>).toJSON()
  expect(tree).toMatchSnapshot();
})

export default {};

/*
import { useParams } from 'react-router-dom';
import { gql, useQuery } from '@apollo/client';
import { ReactDOM } from 'react';
import DisplaySingleMovie from '../components/DisplayMovie';
import { render,  cleanup } from '@testing-library/react'; 
import "jest-dom/extend-expect";

afterEach(cleanup);



it("renders without crashing", () => {
   const div = document.createElement("div");
   ReactDOM.render(<DisplaySingleMovie((poster_path="/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg", original_language="English", title="Godzilla vs. Kong", runtime=113, genres=["Action"]))></DisplaySingleMovie>, div)
});

it('renders DisplayMovie correctly', () => {
  const {getByTestId} = render(<DisplaySingleMovie(poster_path="/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg", original_language="English", title="Godzilla vs. Kong", runtime=113, genres=["Action"])></DisplaySingleMovie>)
  expect(getByTestId('')).toHaveTextContent()
})
*/

/*
const GET_MOVIE = gql`
  query Query($id: Int!) {
    movieByID(id: $id) {
      title
    }
  }
`;

const movieID = 412656;

const { loading, error, data } = useQuery(GET_MOVIE, {
    variables: { id: movieID },
  });

test('get title', () => {
    expect(data.GET_MOVIE.title).toBe("Chaos Walking");
});

*/