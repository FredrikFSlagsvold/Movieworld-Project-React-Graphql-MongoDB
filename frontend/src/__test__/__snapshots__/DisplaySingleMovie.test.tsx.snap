// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when  1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "fontFamily": "Verdana, sans-serif, Areal",
      "fontSize": "12px",
      "height": 320,
      "width": 190,
    }
  }
>
  <img
    height="284.8px"
    src="https://image.tmdb.org/t/p/original//pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg"
    width="190px"
  />
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div>
      <strong>
        Godzilla vs. Kong
      </strong>
    </div>
  </div>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div>
      Action
    </div>
  </div>
</div>
`;
